climada/entity/exposures/crop_production.py

```diff
@@ -481,180 +490,201 @@ def init_full_exposure_set(input_dir=INPUT_DIR, filename=None, hist_mean_dir=HIS
     # create exposures for all crop-irrigation combinations and save them
     filename_list = list()
     output_list = list()
-    for i, _ in enumerate(filenames):
-        item = filenames[i].split('_')
-        cropyield = CropyieldIsimip()
-        cropyield.set_from_single_run(input_dir=input_dir, filename=filename,
-                                      hist_mean=hist_mean_dir, bbox=bbox,
-                                      yearrange=yearrange, crop=((item[2]).split('-'))[0],
-                                      irr=((item[2]).split('-'))[1], unit=unit)
-        filename_saveto = ('cropyield_isimip_' + ((item[2]).split('-'))[0] + '-'
-                           + (((item[2]).split('-'))[1]).split('.')[0] + '_'
-                           + str(yearrange[0]) + '-' + str(yearrange[1]) + '.hdf5')
-        filename_list.append(filename_saveto)
-        output_list.append(cropyield)
-        cropyield.write_hdf5(os.path.join(output_dir, 'Exposure', filename_saveto))
-
-    if returns == 'filename_list':
+    for file in filenames:
+        histmean_prop = file.split('_')
+        crop_production = CropProduction()
+        crop_production.set_from_single_run(input_dir=input_dir, filename=filename,
+                                            hist_mean=hist_mean_dir, bbox=bbox,
+                                            yearrange=yearrange,
+                                            crop=((histmean_prop[2]).split('-'))[0],
+                                            irr=((histmean_prop[2]).split('-'))[1], unit=unit)
+        filename_list.append(('crop_production_' + ((histmean_prop[2]).split('-'))[0] + '-'
+                              + (((histmean_prop[2]).split('-'))[1]).split('.')[0] + '_'
+                              + str(yearrange[0]) + '-' + str(yearrange[1]) + '.hdf5'))
+        output_list.append(crop_production)
+        crop_production.write_hdf5(os.path.join(output_dir, 'Exposure', filename_list[-1]))
```

Comment:
I think it is better to above make a variable `filename` or `new_filename` or `expo_filename` or `filename_expo` which can then be appended to filenmae_list and used in the write_hdf5.